src/main/java/com/vroozi/service/SolrDataReindexService.java

```diff
@@ -0,0 +1,382 @@
+  }
+
+  private void commitSolrBatch(HttpSolrClient solr, List<SolrInputDocument> solrInputDocumentList)
+      throws IOException, SolrServerException {
+    if (!solrInputDocumentList.isEmpty()) {
+      solr.add(solrInputDocumentList);
+      solr.commit();
+    }
+  }
+
+  private void logFailedBatch(Integer unitId, int start, int recordsPerPage, String stackTrace) {
+    SolrReindexProcess failedSolrReindexProcess = solrReindexProcessRepository
+        .findByProcessTypeAndProcessStateAndUnitId(ProcessType.RETRY_FAILED,
+            ProcessState.UNPROCESSED, unitId);
+    if (failedSolrReindexProcess == null) {
+      failedSolrReindexProcess = new SolrReindexProcess(unitId);
+      failedSolrReindexProcess.setProcessType(ProcessType.RETRY_FAILED);
+      failedSolrReindexProcess.setProcessState(ProcessState.UNPROCESSED);
+    }
+    failedSolrReindexProcess.getFailedRecords()
+        .add(new FailedRecord(start, recordsPerPage, stackTrace));
+    solrReindexProcessRepository.save(failedSolrReindexProcess);
+  }
+
+  /* Sets vendor_name_lowercase field if there is existing vendor_name field present */
+  private boolean setVendorData(boolean recordUpdated, SolrInputDocument solrInputDocument) {
+    String vendorName = null;
+    if (solrInputDocument.get("vendor_name") != null) {
+      vendorName = solrInputDocument.get("vendor_name").getValue().toString();
+    }
+
+    if (StringUtils.isNotBlank(vendorName)) {
+      solrInputDocument.removeField("vendor_name_lowercase");
+      solrInputDocument.addField("vendor_name_lowercase", vendorName);
+      recordUpdated = true;
+    }
+    return recordUpdated;
+  }
+
+  /**
+   * Sets New Material Group Data in solrInputDocument if there is existing data, else adds an error
+   * regarding mat group not found for that particular solrInputDocument.
+   */
+  private boolean setMaterialGroupData(int unitId, List<String> errors,
+      List<MaterialGroupMapping> materialGroupMappings,
+      SolrInputDocument solrInputDocument, String solrDocumentId)
+      throws UnsupportedEncodingException {
+    boolean updated = false;
+    String matGroupLabel;
+    String matGroupInfo;
+    String matGroupCode = null;
+    if (solrInputDocument.get("mat_group") != null) {
+      matGroupCode = solrInputDocument.get("mat_group").getValue().toString();
+    }
+    if (StringUtils.isNotBlank(matGroupCode)) {
+      MaterialGroupMapping materialGroupMapping = CategoryMatcher
+          .findMaterialGroupMappingAgainstItemMatGroup(materialGroupMappings, matGroupCode);
+      if (materialGroupMapping != null) {
+        matGroupLabel = materialGroupMapping.getCompanyLabel();
+        if (StringUtils.isNotBlank(matGroupLabel)) {
+          matGroupInfo = materialGroupMapping.getCompanyCategoryCode() + "|||" + URLEncoder
+              .encode(matGroupLabel, "UTF-8");
+          solrInputDocument.removeField("mat_group_label");
+          solrInputDocument.removeField("mat_group_info");
```

Comment:
Why these two remove statements aren't part of `removeCopyFields()` method

Reply:
I remember this was introduced after I ran script again and again, for those records which have been updated with these new fields, so removing these two lines will be a better Idea, removing it